pkg/sdk/validation/policy/policy.go

```diff
@@ -119,15 +119,28 @@ func (v *Validator) hasImplAdditionalInputParams(impl gqlpublicapi.Implementatio
 	return true
 }
 
-func (v *Validator) validationResultForTIMetadata(tis []metadata.TypeInstanceMetadata) validation.Result {
-	if len(tis) == 0 {
-		return validation.Result{}
-	}
-
+func (v *Validator) validationResultForTIMetadata(tis []metadata.TypeInstanceMetadata, rules []policy.RulesForTypeInstance) validation.Result {
 	resultBldr := validation.NewResultBuilder("Metadata for")
 
+	unresolvedIDs := map[string]struct{}{}
 	for _, ti := range tis {
 		resultBldr.ReportIssue(string(ti.Kind), "missing Type reference for %s", ti.String(false))
+		unresolvedIDs[ti.ID] = struct{}{}
+	}
+
+	for _, rule := range rules {
+		if _, unresolved := unresolvedIDs[rule.Backend.ID]; unresolved {
+			continue // Type was not resolved, so `ExtendsHubStorage` has zero value
+		}
+
+		if rule.Backend.ExtendsHubStorage {
+			continue
+		}
+		ref := metadata.TypeInstanceMetadata{
+			ID:          rule.Backend.ID,
+			Description: rule.Backend.Description,
+		}
+		resultBldr.ReportIssue("BackendTypeInstance", "Type reference %s is not a Hub storage", ref.String(false))
```

Comment:
```suggestion
		resultBldr.ReportIssue(string(ti.Kind), "missing Type reference for %q", ti.String(false))
		unresolvedIDs[ti.ID] = struct{}{}
	}

	for _, rule := range rules {
		if _, unresolved := unresolvedIDs[rule.Backend.ID]; unresolved {
			continue // Type was not resolved, so `ExtendsHubStorage` has zero value
		}

		if rule.Backend.ExtendsHubStorage {
			continue
		}
		ref := metadata.TypeInstanceMetadata{
			ID:          rule.Backend.ID,
			Description: rule.Backend.Description,
		}
		resultBldr.ReportIssue("BackendTypeInstance", "Type reference %q is not a Hub storage", ref.String(false))
```

Reply:
Here it doesn't make sens as we have already a proper formatting done in `ref.String(false)`. With additional quotes, we will print:
```
* Type reference "ID: "00fd161c-01bd-47a6-9872-47490e11f996", description: "Vault TI"" is not a Hub storage
* missing Type reference for "ID: "id3", name: "id-3""
```